File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    numer = (numer1 * denom2) + (numer2 * denom1)
    denom = denom1 * denom2
    
    i = 2
    while i <= denom:
        if numer % i == 0 and denom % i == 0:
            numer = numer / i
            denom = denom / i
            i = 2
        else:
            i += 1
            
    answer = [numer, denom]
    return answer

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int a = 0;
        int b = 0;
        a = (numer1 * denom2) + (numer2 * denom1);
        b = denom1 * denom2;
        
        int i = 2;
        while (i<=b) {
            if (a % i == 0 && b % i == 0) {
                a = a / i;
                b = b / i;
                i = 2;
            } else {
                i = i + 1;
            }
        }
        
        int[] answer = {a, b};
        return answer;
    }
}